KeePassXC-Browser manual

varjolintu opened the pull request:
Still WIP. Suggestions and fixes are welcome.
[chromeIPass manual](https://github.com/pfn/passifox/blob/master/documentation/chromeIPass.md) was used as a base.

Currently the manual is written with features included in KeePassXC 2.4.0.

Progress:
- [x]  Write all sections
- [x]  Add images
- [ ] Add explanation about the popup icons
- [x] Add a help section for software packages (snap, flat etc..) + jails
- [ ]  Proof reading, grammar check etc

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+Table of contents:
+- [1. Functionality](#1-functionality)
+    - [1.1. Accessing redentials](#11-accessing-credentials)
```
`Accessing redentials` → `Accessing credentials`*

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+
+### 1. Functionality
+
```
(Very minor) note that you don't consistent put a space after headers. (See here, where you put a space, but `1.1` and subheaders, where you don't, and `1.2`, …, where you do.)

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+#### 1.1. Access the credentials
+If KeePassXC-Browser detects an input field on a web page, a request is sent to KeePassXC to check if the page URL matches with any entry in the database.
+The received credentials are accesible via multiple ways which are described in the following sections.
```
(Minor again but) "accessible via multiple ways" sounds weird; I'd write "accessible via [a specific menu]", or "accessible in multiple ways" – probably the latter here.

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+The received credentials are accesible via multiple ways which are described in the following sections.
+
+##### 1.1.1. Popup
+The extension icon background is green and it is shown with a question mark. Clicking on the icon opens a popup on where you can choose the credentials to use.
```
"on where" → "in which" or "where"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+</div>
+
+##### 1.1.2. Autocomplete
+Username field supports autocomplete for showing a list of received credentials. If there is only a password field visible on the page, autocomplete will be still available.
```
"Username fields support autocomplete" instead?

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+##### 1.1.2. Autocomplete
+Username field supports autocomplete for showing a list of received credentials. If there is only a password field visible on the page, autocomplete will be still available.
+By clicking the input field, the autocomplete menu is visible and correct username can be selected. Password field will be filled automatically.
```
"and correct username" → "and the correct username". Similarly to above, "The password field" (instead of "Password field")?

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+</div>
+
+##### 1.1.3. Context menu
+Context menu is accesible on any input field, even when no credentials are detected for the page.
```
"The context menu"? Also "accesible" → "accessible"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+The following options are available:
+
+**Fill User & Pass** will only work if KeePassXC-Browser received only one pair of credentials. Otherwise it shows you a notification and autocomplete or popup should be used instead.
+**Fill Pass only** will work similary, but fills only the password of a single pair of credentials.
```
"similary" → "similarly"
For this and the above line, I'd make the tense into "**Fill User & Pass** only works if…" and "**Fill Pass only** works similarly…", like the "**Fill TOTP**" entry (i.e. "works" instead of "will work").

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+**Fill Pass only** will work similary, but fills only the password of a single pair of credentials.
+**Fill TOTP** works after input fields are already filled with credentials, and fills a Time-based One-time password from the selected credentials entry.
+**Show Password Generator icons** restarts the detection of *visible* password fields on the page and adds the key-icon for the password generator to each of them.
+**Save credentials** triggers the credential saving popup and allows user to create new entry to the database or update an existing one.
```
"create new entry to the database" → "create a new entry in the database"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+src="{{ site.baseurl }}/images/browser-migration/manual_custom_credential_fields.png">
+</div>
+
+Additional fields called *String Fields* can be selected before the confirm. This functionality is explained in [1.7. Fill additional values using String Fields](#17-fill-additional-values-using-string-fields).
```
"before the confirm" → "before the confirmation"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+### 1.5. Create new or modify existing credentials
+
+KeePassXC-Browser is able to detect form submits. If unsaved or modified credentials are detectd, the KeePassXC-Browser icon starts blinking red.
```
"form submits" → "form submissions"
"detectd" → "detected"

varjolintu replied:
I really need a new keyboard..

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+KeePassXC-Browser is able to detect form submits. If unsaved or modified credentials are detectd, the KeePassXC-Browser icon starts blinking red.
+When clicked, it remains red until you add, update or dismiss the detected changed. It is also possible to put the site to an ignore list when no 
+credential changes are informed in the future.
```
"when no credential changes are informed in the future" → "so that no credential changes are detected in the future"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+src="{{ site.baseurl }}/images/browser-migration/manual_remember_credentials_popup.png">
+</div>
+
+If you have multiple credentials for a page, clicking *Update* opens a list of all available entries on which you can select the outdated entry.
```
"on which" → "from which"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+In *Settings -> General* the following settings are related to this feature:
+- *Blink Time* (The maximum time icon blinks after detecting a credential change)
+- *Redirect Offset* (The time icon blinks after a page redirect)
```
"The time icon blinks after" → "The time the icon blinks for after"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+- *Show a notification when new credentials can be saved to the database.*
+- *Save domain only.* (When saving new credentials save only the domain instead of full URL)
+
+Please notice that detecting the form changes might not work with AJAX requests.
```
"Please notice" → "Please note"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+### 1.6. Choose credentials for HTTP Basic Auth requests
+
+KeePassXC-Browser can try automatically fill HTTP Basic Auth dialogs. If only one credentials are available, the dialog will be filled automatically.
```
"try automatically fill" → "try to automatically fill"
"one credentials are" → "one set of credentials is"?

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+### 1.7. Fill additional values using String Fields
+
+It is possible to fill additional information by defining String Fields to a page.
```
"to a page" → "on a page"?

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+It is possible to fill additional information by defining String Fields to a page.
+
+1. Use *Choose custom credential fields for this page* from the popup. Username and password fields can be skipped if not needed.
+2. Now additional fields called String Fields can be choosed. Dropdown elements can be choosed also. Remember the order they are selected.
```
can be choosed → can be chosen

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+### 1.8. Fill TOTP
+
+KeePassXC-Browser can return Time-based One-time Passwords from the database entry. With KeePassXC 2.3.3. and newer no additional attributes are needed for the fill.
```
It looks like the capitalization for TOTP should be "**T**ime-based **O**ne-**T**ime **P**asswords" (i.e. capitalize the "-time" bit)

Throne3d replied:
`2.3.3.` shouldn't have a period after it, I think? "With KeePassXC 2.3.3 and newer"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+
+#### 2.1. Settings: General
+
+On the *General* settings page key features, for example autocomplete, password generator and auto-fill can be enabled or disabled.
```
`On the *General* settings page, key features – such as autocomplete, the password generator and auto-fill – can be enabled or disabled.`

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+Only credentials with a full matching URL will be returned. This is disabled by default.
+
+##### Sort matching credentials by title / username
+Credentials seen in KeePassXC-Browser's autocomplete or popup login dialog are sorted by username as a default. When sorting by title is used, the credentials are first
```
"When sorting by title is used," → "When sorting by title," or "When 'sort by title' is used" (where "sort by title" would be the name of the option in the display)

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+Removes all keys connected with KeePassXC-Browser from KeePassXC.
+
+##### Forget all remembered permissions
+Removes all permissions (Allow/Deny) from entries. When accessing a page with matching credentials, the permissions are asked again.
```
"are asked again" → "are asked for again"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+Allow all credentials without prompting an access confirmation dialog.
+
+##### Never ask before updating credentials
+If a credential update is requested, new username and/or password is updated automatically without a confirmation.
```
"new username and/or password" → "the new username and/or password"

Throne3d commented on docs/keepassxc-browser-manual.md:
```diff
@@ -0,0 +1,333 @@
+Allows KeePassXC to return additional attributes described in section [1.7. Fill additional values using String Fields](#17-fill-additional-values-using-string-fields).
+
+##### Update native messaging manifest files at startup
+Location of KeePassXC or keepassxc-proxy can change, for example when using a portable version of the application. This setting updates the application location to the native
```
"Location of" → "The location of"

varjolintu commented:
All sections are now written and images are added.

varjolintu commented:
@apiraino The wiki page you are referring to is from my old repository. The [migration guide](https://keepassxc.org/docs/keepassxc-browser-migration/) has replaced it.

varjolintu commented:
Added new images (some of the settings were moved to Database settings instead of global Browser Integration settings). Also added a few workarounds for software packages.

droidmonkey commented:
@varjolintu I think we can close this with our new documentation becoming available.
